pkg/identity/identity.go

```diff
@@ -14,10 +14,42 @@
 
 package identity
 
-import "encoding/asn1"
+import (
+	"encoding/asn1"
+	"errors"
+	"strconv"
+	"strings"
+)
 
 // OIDMatcher holds an OID field and a list of values to match on
 type OIDMatcher struct {
 	ObjectIdentifier asn1.ObjectIdentifier `yaml:"objectIdentifier"`
 	ExtensionValues  []string              `yaml:"extensionValues"`
 }
+
+// CustomOID holds an OID field represented in dot notation and a list of values to match on
+type CustomExtension struct {
+	ObjectIdentifier string   `yaml:"objectIdentifier"`
+	ExtensionValues  []string `yaml:"extensionValues"`
+}
+
+// parseObjectIdentifier parses a string representing an ObjectIdentifier in dot notation
+// and converts it into an asn1.ObjectIdentiifer.
+func ParseObjectIdentifier(oid string) (asn1.ObjectIdentifier, error) {
```

Comment:
Nit: let's make sure the docstring and the function have the same capitalization (exposed or not)